### PR TITLE
Enhance error message when trying to access storage var without `.self` prefix

### LINKED ISSUE #261: Improve error message when the user forgets to use `self` to refer to storage vars
In case one drops `self.` when referring to the `var` storage variable, we should suggest to use `self`.

### PR DESCRIPTION
**Pull request by Gusarich**
Closes #261 

- [x] I have updated CHANGELOG.md
- [ ] ~~I have documented my contribution in Tact Docs: https://github.com/tact-lang/tact-docs/pull/PR-NUMBER~~
- [x] I have added tests to demonstrate the contribution is correctly implemented: this usually includes both positive and negative tests, showing the happy path(s) and featuring intentionally broken cases
- [x] I have run all the tests locally and no test failure was reported
- [x] I have run the linter, formatter and spellchecker
- [x] I did not do unrelated and/or undiscussed refactorings

### CODE REVIEW

**Comment by Gusarich** [comment]
@anton-trunov what if we try to find the variable name in all of the structs in the current context, not only `self`?

**Comment by anton-trunov** [comment]
Let's add similar tests for traits

**Comment by anton-trunov** [comment]
> what if we try to find the variable name in all of the structs in the current context, not only self?

hmmm, not sure about this one, because it might return several results (different struct types can have identical field names)

**Comment by Gusarich** [comment]
> Let's add similar tests for traits

done

File: src/types/stmts-failed/var-does-not-exist3.tact [hunk 1]
```diff
@@ -0,0 +1,19 @@
+primitive Int;
+
+trait BaseTrait { }
+
+trait TestTrait {
+    someField: Int;
+}
+
+contract Test with TestTrait {
```
Comment (anton-trunov) [inline src/types/stmts-failed/var-does-not-exist3.tact:9]: I actually meant something like this:

```
trait Ownable {
    owner: Address;

    fun requireOwner() {
        nativeThrowUnless(132, sender() == owner);
    }
}
```

**Review by anton-trunov — APPROVED** [review]

**Pull request merged**